<?xml version="1.0" encoding="UTF-8" standalone="yes" ?>
<!DOCTYPE bugzilla SYSTEM "https://bugzilla.redhat.com/page.cgi?id=bugzilla.dtd">

<bugzilla version="4.4.12068.1"
          urlbase="https://bugzilla.redhat.com/"
          
          maintainer="bugzilla-error-list@redhat.com"
>

    <bug>
          <bug_id>1374491</bug_id>
          
          <creation_ts>2016-09-08 16:10:00 -0400</creation_ts>
          <short_desc>ansible-node group for dnf does not contain libselinux-python</short_desc>
          <delta_ts>2016-09-08 16:17:19 -0400</delta_ts>
          <reporter_accessible>1</reporter_accessible>
          <cclist_accessible>1</cclist_accessible>
          <classification_id>2</classification_id>
          <classification>Fedora</classification>
          <product>Fedora</product>
          <component>ansible</component>
          <version>24</version>
          <rep_platform>All</rep_platform>
          <op_sys>Linux</op_sys>
          <bug_status>CLOSED</bug_status>
          <resolution>RAWHIDE</resolution>
          
          
          <bug_file_loc></bug_file_loc>
          <status_whiteboard></status_whiteboard>
          <keywords></keywords>
          <priority>unspecified</priority>
          <bug_severity>medium</bug_severity>
          <target_milestone>---</target_milestone>
          
          
          <everconfirmed>1</everconfirmed>
          <reporter name="Steve Kuznetsov">skuznets</reporter>
          <assigned_to name="Stephen Gallagher">sgallagh</assigned_to>
          <cc>a.badger</cc>
    
    
    <cc>athmanem</cc>
    
    
    <cc>kevin</cc>
    
    
    <cc>kupo</cc>
    
    
    <cc>mark</cc>
    
    
    <cc>maxim</cc>
    
    
    <cc>toromoti</cc>
          <qa_contact name="Fedora Extras Quality Assurance">extras-qa</qa_contact>
          
          <cf_fixed_in></cf_fixed_in>
          <cf_doc_type>If docs needed, set a value</cf_doc_type>
          <cf_release_notes></cf_release_notes>
          <cf_story_points>---</cf_story_points>
          
          <cf_environment></cf_environment>
          <cf_last_closed>2016-09-08 16:17:19</cf_last_closed>
          <cf_type>Bug</cf_type>
          <cf_regression_status>---</cf_regression_status>
          <cf_mount_type>---</cf_mount_type>
          <cf_documentation_action>---</cf_documentation_action>
          <cf_crm></cf_crm>
          <cf_verified_branch></cf_verified_branch>
          <cf_category>---</cf_category>
          <cf_ovirt_team>---</cf_ovirt_team>
          
          <cf_cloudforms_team>---</cf_cloudforms_team>
          
          
          
          
          <target_release>---</target_release>
          
          <votes>0</votes>

      

      

      

          <comment_sort_order>oldest_to_newest</comment_sort_order>  
          <long_desc isprivate="0" >
    <commentid>9692070</commentid>
    <comment_count>0</comment_count>
      <attachid>1199193</attachid>
    <who name="Steve Kuznetsov">skuznets</who>
    <bug_when>2016-09-08 16:10:40 -0400</bug_when>
    <thetext>Created attachment 1199193
sample playbook

Description of problem:
The `ansible-node` group, which is created to try to contain all of the dependencies necessary to get a remote host to be a valid target for Ansible, does not contain `libselinux-python`, which is a necessary dependency if the target host has SELinux.

How reproducible:
Run a playbook against a Fedora 24 host with the `ansible-node` group but without `libselinux-python`.

Steps to Reproduce:
# on remote F24 host
$ dnf group install -y ansible-node
# locally
$ cat &lt;&lt;EOF &gt;playbook.yml
---
- name: try to run a playbook without SELinux bindings
  hosts: hostname
  become: yes
  become_method: sudo
  become_user: root

  tasks:
    - lineinfile:
        dest: /etc/yum.conf
        line: &apos;# comment&apos;
        state: present
        create: true
EOF
$ cat &lt;&lt;EOF &gt;inventory
hostname
EOF
$ ansible-playbook -i inventory playbook.yml

Actual results:
PLAY [try to run a playbook without SELinux bindings] **************************

TASK [setup] *******************************************************************
ok: [openshiftdevel]

TASK [lineinfile] **************************************************************
fatal: [openshiftdevel]: FAILED! =&gt; {&quot;changed&quot;: false, &quot;failed&quot;: true, &quot;msg&quot;: &quot;Aborting, target uses selinux but python bindings (libselinux-python) aren&apos;t installed!&quot;}

NO MORE HOSTS LEFT *************************************************************
	to retry, use: --limit @playbook.retry

PLAY RECAP *********************************************************************
openshiftdevel             : ok=1    changed=0    unreachable=0    failed=1   



Expected results:
PLAY [try to run a playbook without SELinux bindings] **************************

TASK [setup] *******************************************************************
ok: [openshiftdevel]

TASK [lineinfile] **************************************************************
changed: [openshiftdevel]

PLAY RECAP *********************************************************************
openshiftdevel             : ok=2    changed=1    unreachable=0    failed=0</thetext>
  </long_desc><long_desc isprivate="0" >
    <commentid>9692075</commentid>
    <comment_count>1</comment_count>
    <who name="Stephen Gallagher">sgallagh</who>
    <bug_when>2016-09-08 16:13:49 -0400</bug_when>
    <thetext>Patch submitted to comps: https://pagure.io/fedora-comps/pull-request/43</thetext>
  </long_desc><long_desc isprivate="0" >
    <commentid>9692081</commentid>
    <comment_count>2</comment_count>
    <who name="Kevin Fenzi">kevin</who>
    <bug_when>2016-09-08 16:17:19 -0400</bug_when>
    <thetext>Merged. ;)</thetext>
  </long_desc>
      
          <attachment
              isobsolete="0"
              ispatch="0"
              isprivate="0"
          >
            <attachid>1199193</attachid>
            <date>2016-09-08 16:10:00 -0400</date>
            <delta_ts>2016-09-08 16:10:40 -0400</delta_ts>
            <desc>sample playbook</desc>
            <filename>playbook.yml</filename>
            <type>text/x-vhdl</type>
            <size>263</size>
            <attacher name="Steve Kuznetsov">skuznets</attacher>
            
              <data encoding="base64">LS0tCi0gbmFtZTogdHJ5IHRvIHJ1biBhIHBsYXlib29rIHdpdGhvdXQgU0VMaW51eCBiaW5kaW5n
cwogIGhvc3RzOiBob3N0bmFtZSAKICBiZWNvbWU6IHllcwogIGJlY29tZV9tZXRob2Q6IHN1ZG8K
ICBiZWNvbWVfdXNlcjogcm9vdCAKIAogIHRhc2tzOgogICAgLSBsaW5laW5maWxlOgogICAgICAg
IGRlc3Q6IC9ldGMveXVtLmNvbmYKICAgICAgICBsaW5lOiAnIyB3aGF0ZXZlcicKICAgICAgICBz
dGF0ZTogcHJlc2VudAogICAgICAgIGNyZWF0ZTogdHJ1ZQo=
</data>

          </attachment>
      

    </bug>

</bugzilla>